src/traces/box/defaults.js

```diff
@@ -40,9 +40,14 @@ function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
             if(sd && sd.length) boxmeanDflt = 'sd';
         }
     }
-    coerce('boxmean', boxmeanDflt);
 
     coerce('whiskerwidth');
+    var sizemode = coerce('sizemode');
+    if(sizemode === 'quartiles') {
+        coerce('boxmean', boxmeanDflt);
+    }
+    coerce('showwhiskers', sizemode === 'quartiles');
+    coerce('sdmultiple');
```

Comment:
Made this change. Guess you're saying performance of 2 if's >> coerce

Reply:
> Made this change

Pushed? I don't see it. Anyway it's not mainly about performance although performance will benefit. The goal is that if an attribute is not meaningful (based on other attributes) it should not be coerced at all so that it does not exist in `_fullData` (or `_fullLayout`). This helps with logic for drawing, as then `_fullData` is a precise description of what should be drawn, and it's also used by some external packages like [`react-chart-editor`](https://github.com/plotly/react-chart-editor) to decide what controls to display (if a control won't do anything it should not be shown).

Reply:
Sorry just pushed